integration_tests/integration_test.go

```diff
@@ -260,15 +261,200 @@ var _ = Describe("Tablecloth HTTP listener", func() {
+			})
+		})
+
+		Context("the new server exits shortly after starting", func() {
+			It("should continue running the old server", func() {
+				resp, err := http.Get("http://127.0.0.1:8081/")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(resp.StatusCode).To(Equal(200))
+				firstBody, _ := ioutil.ReadAll(resp.Body)
+				resp.Body.Close()
+
+				Expect(os.Remove(cwd + "/test_servers/current")).To(Succeed())
+				Expect(os.Symlink(cwd+"/test_servers/errorer", cwd+"/test_servers/current")).To(Succeed())
+
+				withSilentOutput(func() {
+					reloadServer(serverCmd)
+				})
+
+				resp, err = http.Get("http://127.0.0.1:8081/")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(resp.StatusCode).To(Equal(200))
+				newBody, _ := ioutil.ReadAll(resp.Body)
+				resp.Body.Close()
+
+				Expect(newBody).To(Equal(firstBody))
+			})
+
+			if canReadProcessFds() {
+				It("should not leak file descriptors when reloading fails", func() {
+					resp, err := http.Get("http://127.0.0.1:8081/")
+					Expect(err).NotTo(HaveOccurred())
+					Expect(resp.StatusCode).To(Equal(200))
+
+					initalFds := getProcessFds(serverCmd)
+
+					Expect(os.Remove(cwd + "/test_servers/current")).To(Succeed())
+					Expect(os.Symlink(cwd+"/test_servers/errorer", cwd+"/test_servers/current")).To(Succeed())
+
+					withSilentOutput(func() {
+						reloadServer(serverCmd)
+					})
+
+					resp, err = http.Get("http://127.0.0.1:8081/")
+					Expect(err).NotTo(HaveOccurred())
+					Expect(resp.StatusCode).To(Equal(200))
+
+					currentFds := getProcessFds(serverCmd)
+					Expect(currentFds).To(Equal(initalFds))
+				})
+			} else {
+				PIt("leaking file descriptors test requires /proc/<pid>/fd directories")
+			}
+
+			It("should successfully handle subsequent reload requests with a good server", func() {
+				resp, err := http.Get("http://127.0.0.1:8081/")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(resp.StatusCode).To(Equal(200))
+
+				Expect(os.Remove(cwd + "/test_servers/current")).To(Succeed())
+				Expect(os.Symlink(cwd+"/test_servers/errorer", cwd+"/test_servers/current")).To(Succeed())
+
+				withSilentOutput(func() {
+					reloadServer(serverCmd)
+				})
+				resp, err = http.Get("http://127.0.0.1:8081/")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(resp.StatusCode).To(Equal(200))
+
+				// Now point back at a good server
+				Expect(os.Remove(cwd + "/test_servers/current")).To(Succeed())
+				Expect(os.Symlink(cwd+"/test_servers/v2", cwd+"/test_servers/current")).To(Succeed())
+
+				reloadServer(serverCmd)
+
+				resp, err = http.Get("http://127.0.0.1:8081/")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(resp.StatusCode).To(Equal(200))
+
+				body, _ := ioutil.ReadAll(resp.Body)
+				Expect(string(body)).To(ContainSubstring("Hello from v2"))
+			})
+		})
+	})
 })
 
+var serverOutputWriter = &struct{ io.Writer }{os.Stderr}
+
+func withSilentOutput(f func()) {
+	serverOutputWriter.Writer = ioutil.Discard
```

Comment:
It's also printed when running `go test` without any params in the integration_test directory.

Note: this only silences the output for very specific periods, while reloading the server when we expect it to output warning messages.  I felt this was appropriate to avoid expected warnings polluting the test output.